compiler/rustc_privacy/src/lib.rs

```diff
@@ -988,7 +988,10 @@ impl<'tcx> Visitor<'tcx> for NamePrivacyVisitor<'tcx> {
     fn visit_expr(&mut self, expr: &'tcx hir::Expr<'tcx>) {
         if let hir::ExprKind::Struct(qpath, fields, ref base) = expr.kind {
             let res = self.typeck_results().qpath_res(qpath, expr.hir_id);
-            let adt = self.typeck_results().expr_ty(expr).ty_adt_def().unwrap();
+            let Some(adt) = self.typeck_results().expr_ty(expr).ty_adt_def() else {
```

Comment:
is this error case only reachable for `ty::Error`? If so, I'm wondering if we should return `Resut<Option<AdtDef>, ErrorGuaranteed>`. The churn may be useless, or it exposes more such cases

Reply:
The result of `self.typeck_results().expr_ty(expr)` is `ty::Error`. I don't know if that's the only thing that can reach here; it's the only thing we've seen reach here because this is the first time the the adt_def has ever failed.